libp2p/dialer.nim

```diff
@@ -195,6 +195,10 @@ proc internalConnect(
     try:
       self.connManager.storeMuxer(muxed)
       await self.peerStore.identify(muxed)
+      asyncSpawn self.connManager.triggerPeerEvents(
```

Comment:
why `asyncSpawn`?

Reply:
Because we do not know what's the callback will be. It could possibly be an infinite loop for example. AsyncSpawn is safer to avoid this kind of behavior.

Reply:
Is it? Running an infinite loop in the background doesn't sound like a good alternative. How about we use `await` with a timeout?

Reply:
What I mean by that is that the peerEvent is public, we do not control what libp2p user could do here.
Another point to consider is that this `triggerPeerEvent(Identified)` is called in important part of the code. Let's take a case where the callback do a lot of computation (let's say 200ms), every time we dial a peer or we accept a connection, 200 ms is spent waiting for the callback to finish before anything else can be done.
And last point, if you look at how the other events are called, they also use asyncSpawn (https://github.com/vacp2p/nim-libp2p/blob/master/libp2p/connmanager.nim#L310-L317)

Reply:
I'm still not convinced. AFAIK, chronos is single-threaded and once this task is executed it won't yield control until it stops running. It's not executed in parallel.

Reply:
In the case of the scenario of 200 milliseconds it'll slow down every dial by this amount. I think that it's really preferable to give the newly created connection to the user directly and await for these after the dial (or accept) is completed.

Reply:
After a private conversation, it's indeed better to not use asyncSpawn here :+1: 
It's now change to `await` instead